src/error.ts

```diff
@@ -0,0 +1,27 @@
+import chalk from 'chalk'
+
+// OpError stands for operation error.
+// it should be used to show a friendly error mesage after an operation is failed.
+// use it with throw to benefit from oclif's error handling. it print the error
+// and exit with 1.
+export class OpError {
```

Comment:
I don't understand how this works. Can't we just keep the `this.error` that is standard in this framework instead of doing some complicated magic stuff

Reply:
this.error does not print friendly errors because it dumps the stack with the error message. with this implementation, we only show the error string in red color, like in the Go cli.

OpError is an error type that it implements the error format accepted by the cli/cli's error package that we use. it seems a bit tricky but definitely not an anti-pattern. Otherwise we can create an another func that prints the error and exists with 1 afterwards, they both are the same.

If we use `require()`, same thing, the cli dumps the stack as well, which is not that user friendly.

As a conclusion, `Error` stands for un expected errors like the `panic()` in Go and `OpError` stands for showing user friendly error messages. They both exit with 1 but `OpError` doesn't print the stack and has red coloring.

Reply:
from the documentation `this.error` is displaying a friendly error message https://oclif.io/docs/commands#thiserrormessage-string-error-options-code-string-exit-number so I would just use what the framework is giving especially that they manage also the case in DEBUG

Reply:
my bad, I tested again and `this.error` work just fine and shows friendly messages